services/monster_service.go

```diff
@@ -0,0 +1,36 @@
+package services
+
+import (
+	"fmt"
+
+	"github.com/Le-MaliX/ACADEMY-GO-Q42021/domain/model"
+	"github.com/Le-MaliX/ACADEMY-GO-Q42021/interface/repository"
+)
+
+func GetAllMonsters() ([]model.Monster, error) {
+	monsters, err := repository.GetMonstersData()
+	if err != nil {
```

Comment:
It's great that you're handling your errors, but we have to take some details into consideration. For errors that the application generates, like errors reading files, casting errors, etc... you would like to log those errors, this will allow you to take a look at them if something fails as a dev.

And then after logging the errors, you want to return to the user or client of your API, something more user-friendly, you don't want to disclose the details of how your app works. What errors to exactly return to the user is up to your own criteria, but for example:

if your app fails on stablishing a connection to the DB, then you don't want to tell the user that you weren't able to connect to the DB, maybe we could return something like an InternalServerError with a message that goes "Something went wrong", while logging the actual DB error we got.

Doing this sort of more customized error messages would also imply some reworking on how you handle your errors, specifically on the controller layer. There you could actually check what type of error you got to decide what error and message to return to the user, since you don't always want to return a bad request for every error that could happen.

I hope I was clear enough, if not please let me know!